### PR TITLE
mono::ir::Expr::UnionFieldPtrAtIndex -> GetElementPointer

### LINKED ISSUE #5980: Generalise mono::ir::Expr::UnionFieldPtrAtIndex to GetElementPointer for Structs
Proposed by @folkertdev,
UnionFieldPtrAtIndex works on TagUnions. It takes a tag Id and an index. So selecting `d` from the tag union `[A, B a b, C c d e]` is done like this: `UnionFieldPtrAtIndex 2 1 MyTagUnion`. The new way of usage would look like this:
`GetElementPointer [ 2, 1] MyTagUnion` 

This function needs to generalise over a record payload. Selecting the `rest` member from this type
`[ Cons { x: { first: a, rest: ConsList a }, y : Bool }, Nil ]` 
would look like this: 
`GetElementPointer [ 0, 0, 1] MyTagUnion` 
where [0, 0, 1] says *0th tag id, index 0 member = x, index 1 member = rest*

### PR DESCRIPTION
**Pull request by HajagosNorbert**
Now GetElementPointer can do everything UnionFieldPtrAtIndex was able to do. A step towards #5980

### CODE REVIEW

**Comment by Anton-4** [comment]
Feel free to send me a PM on zulip whenever you're ready for CI tests to run :)

File: crates/compiler/mono/src/tail_recursion.rs [hunk 1]
```diff
@@ -907,13 +907,13 @@ impl<'a> TrmcEnv<'a> {
                                 reuse: None,
                             };
 
-                            let let_tag = |next| Stmt::Let(*symbol, tag_expr, *layout, next);
+                            let indices = vec![in env.arena; cons_info.tag_id as u64, recursive_field_index as u64].into_bump_slice();
```
Comment (folkertdev) [inline crates/compiler/mono/src/tail_recursion.rs:910]: using `arena.alloc([cons_info.tag_id as u64, recursive_field_index as u64])` is slightly more efficient here

**Review by folkertdev — COMMENTED** [review]
Looking good, one minor nit

**Review by folkertdev — APPROVED** [review]

**Comment by HajagosNorbert** [comment]
> just as a note, ...

Yes, the big button was too tempting. I'll remember that, thanks!

**Pull request merged**